### ISSUE TITLE
could not get media by media_id and user_id from database: record not found in home page

### ISSUE THREAD

**Issue by JordanDIG** [photoview/photoview#729]
**Describe the bug**
I just installed Photoview, when I go to the WEB interface I have no photo loaded and this error appears in a loop.

**To Reproduce**
1. install PhotoView

**Expected behavior**
No photo of the reception desk loads

**Screenshots**
[https://share.jordand.fr/yUPI7/JoneQiMI00.png](https://share.jordand.fr/yUPI7/JoneQiMI00.png)

**Your environment:**
Feel free to remove and add what you think might be useful.

 - Are you using Docker? [yes ]
 - Database [mariadb 10.5]
 - Browser [chrome]
 - Version [2.3.13]

**Comment by Shmik2003** [comment]
You can try to open Photoview, disable connection to the Internet (or disable Photoview server) and open info about this picture. You will see name of image, it's directory and other information. As I think, this error means that this image aren't scanned yet or don't have preview. I believe, you can keep Photoview active for scanning media and this trouble will resolve itself. A small flaw

**Comment by JordanDIG** [comment]
> You can try to open Photoview, disable connection to the Internet (or disable Photoview server) and open info about this picture. You will see name of image, it's directory and other information. As I think, this error means that this image aren't scanned yet or don't have preview. I believe, you can keep Photoview active for scanning media and this trouble will resolve itself. A small flaw

I have already tried to wait 24 hours but the problem is still there...

**Comment by bigjimmy546** [comment]
I have the same issue, problem persists after done scanning.

Are you using Docker? [yes ]
Database [mariadb 10.5]

**Comment by JordanDIG** [comment]
> I have the same issue, problem persists after done scanning.
> 
> Are you using Docker? [yes ] Database [mariadb 10.5] Release Version 2.3.13 ([[446ef9a](https://github.com/photoview/photoview/commit/446ef9a46485bd18bb1c6890d485344786a4e90c)]

I have migrate to Photoprism but yes I was on this version

**Comment by hugoamvieira** [comment]
+1 on this issue. Currently running the `edge` docker image, and also tried completely deleting the database and redoing the entire library from scratch, did not help

**Comment by bigjimmy546** [comment]
Switched to photoprism
Love the simplicity of photo view but waiting for the issue to be solved

**Comment by lockheed** [comment]
I just installed latest Photoview and the issue continues. Is this intended to be fixed?

**Comment by kan84** [comment]
I installed 2.3.12 and removed old db but issue still persists

**Comment by fmnamado** [comment]
Has anyone identified why this occurs?
Is some rogue image, is it random, ... ?
I have redone all my library and scanning was going OK but now this bug appeared.

How can I know the version I am using?
Mine says: 

Release Version
edge (d4f30795)
Build date
2022-03-28T16:52:24Z

I tried pulling other version than this, in portainer, changing the tag and selecting recreate and pull image but I can't seem to fix it, can someone point me how can I be sure which version I am running?

Thank you.

**Comment by sloniupl** [comment]
Had exactly the same issue on 2.3.13 using docker compose " image: viktorstrate/photoview:2 "
After reverting back to 2.3.12 and res-caning again everything is working correctly :)

**Comment by fmnamado** [comment]
Which tag version should i use to revert to .12 version?
And in the settings tab, which version will this .12 version display?
Will it repair after just scan again or full scan scratch from zero?

Ty

**Comment by sloniupl** [comment]
In your docket compose just use "image: viktorstrate/photoview:2.3.12"" after that clean cookies for your website and just too a normal scan.

**Comment by fmnamado** [comment]
Hello and ty for your help.

Before doing what you said, I went to see the version, it was as I posted.
Symptoms:
In timeline I could only see blurhashes, if I pressed the I of the photo, it would open the tab on the right, but without seing the photo.
If I went to albums, it appeared NOTHING.

If i create and login with another user, in timeline the photos appear and in albums aswell. But if I press in the i from info, it says on the tab itself
"Error: could not get media by media_id and user_id from database: record not found"
And in the main window, notification like, says:

"Server error
You are being logged out in an attempt to recover. Cannot query field "description" on type "MediaEXIF".
Something went wrong
Server error: Cannot query field "description" on type "MediaEXIF". at (undefined)
Something went wrong
Server error: could not get media by media_id and user_id from database: record not found at (media)
Something went wrong
Server error: could not get media by media_id and user_id from database: record not found at (media)"

Then I went to try what you told.
I changed the tag in my stack and recreated.
I even went to portainer to press in recereate and for IMAGE PULL.

Now it says
"Photoview Version
Release Version

2.3.12 (0b9d04ef)
Build date

2022-02-11T12:07:25Z"

So I assume it is OK.

I did a scan, let it finish and nothing changed.
Same non-appearing images, etc, as I mentioned above.

Any more ideas?

**Comment by sloniupl** [comment]
Tryscan. For all users and than see. It worked for me where on timeline i
had only white images instead of photo and when clicking "i" i had the same
exact error.

**Comment by fmnamado** [comment]
Every time I mentioned "scan" i meant "scan all users", that's the button I always press.

Did you got white images or blurhashes?
Because in timeline I have blurhashes and not white images.

**Comment by kiesekae** [comment]
For anyone else running into this issue where you get some sporadic white thumbnails in your timeline and albums and the error: `Something went wrong
Server error: could not get media by media_id and user_id from database: record not found at (media)`, this happens for some reason when you scan your library from the "Users Level" opposed to clicking the "Scan all users" button. 

Once you run the Scan All Users button, it should fix the issue. 

Hope this helps someone. 
Cheers!

**Comment by eerison** [comment]
> For anyone else running into this issue where you get some sporadic white thumbnails in your timeline and albums and the error: `Something went wrong Server error: could not get media by media_id and user_id from database: record not found at (media)`, this happens for some reason when you scan your library from the "Users Level" opposed to clicking the "Scan all users" button.
> 
> Once you run the Scan All Users button, it should fix the issue.
> 
> Hope this helps someone. Cheers!

I did it, but it does not solve the issue :/

**Comment by jave** [comment]
I also have this.

**Comment by DaCHack** [comment]
For me a full rescan helped. Seems like the scan was interrupted somewhere before which led to incomplete caching and DB entries...

**Comment by ruudfeen** [comment]
I did some research in the code for this problem, the cause of this problem is missing records in the media_urls table which results in 0 records when retrieving the media, I didn't find the cause for the missing records.

**Comment by ruudfeen** [comment]
After some more investigation it seems (at least in my case) that it had problems with corrupted images. It looks like it stops scanning after it fails to process a corrupted image. When I found out which images it struggled with, I deleted them and the directory was scanned correctly after that. It looks like the  software doesn't recover from faulty images. I also couldn't find an easy way to find which image is corrupt.
